Doc/setup (#14)
AutoResearch/autora-template: README.md

pip install -e ".[dev]"
```

*Note: You may install new dependencies via ``pip install packagename`` inside your virtual environment. If those
dependencies are vital to your package, you will have to add them to the ``pyproject.toml`` (see Step 6 of the 
Contribution Guide).*

## Contribution Guide 

### Step 1: Choose Feature Category

First you have to choose **which type of feature** you would like to add to AutoRA. There are four categories of contributions:<br>

> (1)  **Theorist**: An sklearn regressors that returns an interpretable model relating experiment conditions $X$ to 

Reviewer: ```suggestion
(1)  **Theorist**: An sklearn regressor that returns an interpretable model relating experiment conditions $X$ to 
```

(2)  **Experimentalist**: A method that identifies novel experiment conditions $X'$ that yield scientific merit. 
Experimentalists may either be implemented as a *pooler* (generating a pool of novel experiment conditions) or as a 
sampler (selecting from an existing pool of experiment conditions $X$).<br>
*Example: The [Novelty Sampler](https://github.com/AutoResearch/autora-novelty-sampler) selects novel experiment 
conditions $X'$ with respect to a pairwise distance metric applied to existing experiment conditions $X$.*
<br>

(3) **Experiment Runners**: A method that orchestrates the collecting of observations for a given set of 
experiment conditions.<br>
*Example: The [Firebase-Prolific Runner](https://github.com/AutoResearch/autora-experiment-runner-firebase-prolific) 
enables the collection of behavioral data from human participants via web-based experiments hosted on 
[Firebase](https://firebase.google.com/), using a pool of participants registered through
[Prolific](https://www.prolific.co/).* <br>

(3) **Synthetic Data**: A ground-truth model that implements a hypothesized relationship between experimental conditions

> $X$ and observations $Y$. Synthetic models may act as objects of study for which the underlying mechanisms are known, 

Reviewer: are 'observations' and 'observables' interchangeable for us (see Theorist above)? think so in this context, just wanted to be sure

### Step 2: Delete Irrelevant Files

Depending on which feature you want to contribute, you can remove initialization files from all irrelevant features.
You may delete the following initialization files from the template:

- Theorist: ``src/autora/theorist/example_theorist/__init__.py``
- Experimentalist (Pooler): ``src/autora/experimentalist/pooler/example_pooler/__init__.py``
- Experimentalist (Sampler): ``src/autora/experimentalist/sampler/example_sampler/__init__.py``
- Experiment Runner: ``src/autora/experiment_runner/example_runner/__init__.py``
- Synthetic Data: ``src/autora/synthetic/example_data/__init__.py``

In addition, you may delete the following test files from the template:

- Theorist: ``tests/test_theorist_example.py``
- Experimentalist (Pooler): ``tests/test_experimentalist_pooler_example.py``
- Experimentalist (Sampler): ``tests/test_experimentalist_sampler_example.py``

For instance, if you would like to implement an experimentalist sampler, then you may remove all files listed above 
except for 
- ``src/autora/experimentalist/sampler/example_sampler/__init__.py`` and 
- ``tests/test_experimentalist_sampler_example.py``.

### Step 3: Implement Your Code

> You may now add you a folder in the respective feature category. For instance, if you would like to implement

Reviewer: ```suggestion
You may now add a folder in the respective feature category. For instance, if you would like to implement
```

pip install -e ".[dev]"
```

### Step 7: Publish Your Package

> Once your project is implemented, you may publish it as subpackage of AutoRA. No worries though, in case you are 

Reviewer: ```suggestion
Once your project is implemented, you may publish it as subpackage of AutoRA. If you have not thoroughly vetted your project or would otherwise like to refine it further, you may 
```

> nervous about the state of your package–you will be able to publish it as a pre-release, indicating to users that

Reviewer: ```suggestion
publish it as a pre-release, indicating to users that
```
